lib/Interpreter/Interpreter.cpp

```diff
@@ -356,95 +365,97 @@ namespace cling {
     // itself? One could use a macro (simillar to __dso_handle) to block
     // assignemnt and get around the mangling issue.
     const char* Linkage = LangOpts.CPlusPlus ? "extern \"C\"" : "";
-    if (!NoRuntime && !SyntaxOnly) {
+    if (!NoRuntime) {
       if (LangOpts.CPlusPlus) {
         Strm << "#include \"cling/Interpreter/RuntimeUniverse.h\"\n"
                 "namespace cling { class Interpreter; namespace runtime { "
-                "Interpreter* gCling=(Interpreter*)" << thisP << "; }}\n";
+                "Interpreter* gCling";
+        AssignIfValue(Strm, ThisP, "Interpreter*", " }}");
```

Comment:
Going off the original version, which did this:
```
    if (!isInSyntaxOnlyMode()) {
      // Set up the gCling variable if it can be used
      initializer << "namespace cling {namespace runtime { "
        "cling::Interpreter *gCling=(cling::Interpreter*)"
        << "0x" << std::hex << (uintptr_t)this << " ;} }";
    }
```

I don't know why it did not set the value before, overall it was wrong as gCling was unavailable with **-fsyntax only**.

But in https://github.com/root-project/cling/pull/127 there was an issue with ROOT in `rootling_impl.cxx` using **-fsyntax-only**, and when gCling was set to a value it was barking loudly, so I just made it a declaration in syntax only mode.